crates/ruma-state-res/src/room_version.rs

```diff
@@ -10,6 +12,8 @@ pub enum RoomDisposition {
     Unstable,
 }
 
+#[derive(Debug)]
+#[allow(clippy::exhaustive_enums)]
 pub enum EventFormatVersion {
```

Comment:
Why not make this one non-exhaustive?

Reply:
So I wasn't sure about this or below but my idea was there this would always be a breaking change in the spec, and that it might be better to have an obvious break from version to version :shrug:

Reply:
The whole purpose of room versions is being able to change things without breaking existing code. Why would adding to this inherently be a breaking change?

Reply:
Would the spec not semver bump? I guess I'm not sure about ruma's semver guarantees with respect to the matrix spec changes and their versioning? This probably kinda relates to my question about non_exhaustive enums with a `_Custom` variant and forward compat? If I had a better idea of how/when we want to make breaking changes.